plugins/webhooks/index.js

```diff
@@ -300,14 +305,16 @@ async function createPREvents(options, request) {
 
     scmConfig.prNum = prNum;
 
-    for (let i = 0; i < pipelines.length; i += 1) {
-        const p = pipelines[i];
-        /* eslint-disable no-await-in-loop */
+    const eventConfigs = await Promise.all(pipelines.map(async (p) => {
         const b = await p.branch;
         // obtain pipeline's latest commit sha for branch specific job
-        const configPipelineSha = await pipelineFactory.scm.getCommitSha(scmConfig);
-        /* eslint-enable no-await-in-loop */
+        let configPipelineSha = '';
 
+        try {
+            configPipelineSha = await pipelineFactory.scm.getCommitSha(scmConfig);
+        } catch (err) {
+            winston.info(`skip create event for branch: ${b}`);
```

Comment:
what's the implication of ignoring this error ?, do we need to differentiate between branch does not exist vs any error in talking to git ?

Reply:
It intend to ignore 'branch does not exist'. And 5xx error should not be ignored, so I fixed.